### PR TITLE
export: supress spans in XHTML export

### PR DESCRIPTION
**Pull request by diegoviola**
Fixes: #1445

### CODE REVIEW

**Comment by diegoviola** [comment]
Unfortunately this creates a regression with #1090 which I still couldn't manage to fix. Suggestions?

**Comment by benwbrum** [comment]
_If_ this addresses #1090, then I think we could implement it like this:
```
doc.elements.each("//lb") do |e|
      if preserve_lb
        if e.attributes['break'] == "no"
          # ADD "-" preceding e somehow.  Maybe e.previous_sibling.add_text("-") would work?
        end
        e.replace_with(REXML::Element.new('br'))
      else
        lb = REXML::Element.new('span')
        lb.add_text("")
        lb.add_attribute('class', 'line-break')

        unless e.attributes['break']=="no"
          lb.add_text(' ')
        end
         e.replace_with(lb)
       end      
    end
```

File: app/helpers/abstract_xml_helper.rb [hunk 1]
```diff
@@ -38,23 +43,30 @@ def xml_to_html(xml_text, preserve_lb=true, flatten_links=false)
...
+          if e.attributes['break'] == "no"
+            lb.add_text('')
+          end
+        else
+          if e.attributes['break'] == "no"
+            lb.add_text('-')
+          end
         end
+
+        lb.add_element(REXML::Element.new('br'))
```
Comment (benwbrum) [inline app/helpers/abstract_xml_helper.rb]: If `preserve_lb==false`, we need to stop adding `br` tags, so this line needs to be deleted.

**Comment by benwbrum** [comment]
This gets rid of the `span` tags, but needs a single change to avoid inserting `br` when `preserve_lb==false`.

**Comment by benwbrum** [comment]
We need to do the following:
* Make sure the XHTML export uses `preserve_lb=true`
* Add a single whitespace character when `preserve_lb=false` and the break is hard.

See the following screenshots:
## Original source
See source selected -- note the "hard break" text.
Screenshot from 2019-09-05 16-05-55

## Preserve_LB=*true*
Overview view for a page -- this is perfect.
Screenshot from 2019-09-05 16-06-25

## Preserve_LB=*false*
Multi-page/Read-work view.  See the lack of white space after the hard break.  We need to add some kind of space (probably just a `' '` space character, not an `&nbsp;`) here so that the appearance is `a hard` instead of `ahard`.

Screenshot from 2019-09-05 16-06-48

**Comment by benwbrum** [comment]
This works in all cases now.

**Pull request merged**